### PR TITLE
feat: refactor manifest into more generic approach

### PR DESCRIPTION
**Pull request by nichmor**
Overview:

for pixi global task we need a(nother) pixi-global manifest, which in fact, seems to be a again a toml.

We already have machinery setup to modify and interact with pixi.toml, so I saw an opportunity to extend it in a more abstract way so other type manifests can also use it.

Some things missing:
- [x] public functions docs
- [x] fix clippy warnings
- [x] a way to not clone-that-much

### CODE REVIEW

File: crates/pixi_manifest/src/lib.rs [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
 mod environment;
 mod environments;
 mod error;
 mod feature;
 mod features_ext;
 mod has_features_iter;
 mod has_manifest_ref;
-mod manifest;
+mod manifests;
+// mod manifest;
```
Comment (Hofer-Julian) [inline crates/pixi_manifest/src/lib.rs:12]:
```suggestion
```

**Review by Hofer-Julian — COMMENTED** [review]
> We already have machinery setup to modify and interact with pixi.toml, so I saw an opportunity to extract this in a more generic way so other crates can reuse it.

This gave me the impression that you generalized existing code, but it seems like you added functionality instead of moving existing one.

**Comment by nichmor** [comment]
> > We already have machinery setup to modify and interact with pixi.toml, so I saw an opportunity to extract this in a more generic way so other crates can reuse it.
> 
> This gave me the impression that you generalized existing code, but it seems like you added functionality instead of moving existing one.

changed the wording around it

File: crates/pixi_manifest/src/manifests/mod.rs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+pub mod project;
+
+pub mod manifest;
+
+pub use project::ManifestSource;
+
+use crate::error::TomlError;
+
+/// Represent a wrapper around a TOML document.
```
Comment (Hofer-Julian) [inline crates/pixi_manifest/src/manifests/mod.rs:11]:
```suggestion
/// Represents a wrapper around a TOML document.
```

**Review by Hofer-Julian — APPROVED** [review]

**Review by baszalmstra — COMMENTED** [review]
You moved a number of files into another module (`manifests`) but I think you forgot to also move the snapshots because it looks like were added instead of being moved.

**Comment by tdejager** [comment]
Note that `cargo insta test --all --all-features --unreferenced=delete` is useful.

**Comment by Hofer-Julian** [comment]
> Note that `cargo insta test --all --all-features --unreferenced=delete` is useful.

Sounds like a great candidate for a pixi task :)

**Pull request merged**